### PR TITLE
Restructuring ompl

### PR DESCRIPTION
**Pull request by yimingyanged**
- structure change. before: ompl_solver, ompl_imp_solver. after: ompl_solver
- now instead of using OMPLsolver and specify the algorithm as a parameter, you would directly call the actual algorithms, e.g. RRT. For now, only RRT, RRTConnect and PRM are implemented.
- three statespaces have been implemented: fixed, planar and floating
- added a new srdf of LWR with planar virtual joint for testing
- fixed controlledbasetype in kinematic tree

### CODE REVIEW

File: exotations/solvers/ompl_solver/package.xml [hunk 1]
```diff
@@ -1,25 +1,53 @@
...
+  <!-- Dependencies can be catkin packages or system dependencies -->
+  <!-- Examples: -->
+  <!-- Use build_depend for packages you need at compile time: -->
+  <!--   <build_depend>message_generation</build_depend> -->
+  <!-- Use buildtool_depend for build tool packages: -->
+  <!--   <buildtool_depend>catkin</buildtool_depend> -->
+  <!-- Use run_depend for packages you need at runtime: -->
+  <!--   <run_depend>message_runtime</run_depend> -->
+  <!-- Use test_depend for packages you need only for testing: -->
+  <!--   <test_depend>gtest</test_depend> -->
```
Comment (VladimirIvan) [inline exotations/solvers/ompl_solver/package.xml]: Remove comments

File: exotations/solvers/ompl_solver/package.xml [hunk 2]
```diff
@@ -1,25 +1,53 @@
 <?xml version="1.0"?>
 <package>
```
Comment (VladimirIvan) [inline exotations/solvers/ompl_solver/package.xml]: Use `format="2"`

File: exotations/solvers/ompl_solver/src/ompl_solver/ompl_solver.cpp [hunk 3]
```diff
@@ -0,0 +1,178 @@
+/*
+ * ompl_solver.cpp
+ *
+ *  Created on: 10 Oct 2017
+ *      Author: yiming
+ */
```
Comment (VladimirIvan) [inline exotations/solvers/ompl_solver/src/ompl_solver/ompl_solver.cpp]: Long license text

Comment (VladimirIvan) [reply exotations/solvers/ompl_solver/src/ompl_solver/ompl_solver.cpp]: Have a look at the extra line breaks as well.

**Comment by wxmerkt** [comment]
Lets hold off on merging this until all the collision scene / collision checking is resolved again. I'd like to reduce the number of moving parts and this really requires collision checking to work.

**Comment by wxmerkt** [comment]
Collision checking is now fixed, this should be good to merge.

**Pull request merged**